src/autopas/containers/verletListsCellBased/verletListsCells/traversals/VLCCellPairC08CellHandler.h

```diff
@@ -38,84 +38,83 @@ class VLCCellPairC08CellHandler : public LCC08CellHandler<ParticleCell, Pairwise
-   * @param _soa Structure of Arrays where the particles are loaded if the SoA data layout is used
+   * @param soa Structure of Arrays where the particles are loaded if the SoA data layout is used
    * @param dims number of cells per dimension
    */
   void processCellListsC08(VLCCellPairNeighborList<typename ParticleCell::ParticleType> &neighborList,
                            unsigned long cellIndex, PairwiseFunctor *pairwiseFunctor, DataLayoutOption::Value layout,
-                           SoA<typename ParticleCell::ParticleType::SoAArraysType> *_soa,
+                           SoA<typename ParticleCell::ParticleType::SoAArraysType> *soa,
                            std::array<unsigned long, 3> dims) {
-    auto &aosNeighborList = neighborList.getAoSNeighborList();
-    auto &soaNeighborList = neighborList.getSoANeighborList();
-    auto &globalToLocalIndex = neighborList.getGlobalToLocalMap();
+    const auto &aosNeighborList = neighborList.getAoSNeighborList();
+    const auto &soaNeighborList = neighborList.getSoANeighborList();
+    const auto &globalToLocalIndex = neighborList.getGlobalToLocalMap();
 
-    for (auto &offsets : this->_cellPairOffsets) {
+    // for all interaction pairs defined via the c08 base step
+    for (const auto &[offsetA, offsetB, dirVec] : this->_cellPairOffsets) {
       // the lists are built with a c18 traversal
       // the interaction will always be saved in the smaller cell's neighbor list
-      const auto [offsetCell1, offsetCell2] =
-          std::minmax({cellIndex + std::get<0>(offsets), cellIndex + std::get<1>(offsets)});
+      // std::minmax(a, b) returns references. Hence, we can't use temporaries as arguments.
+      const auto offsetCellA = cellIndex + offsetA;
+      const auto offsetCellB = cellIndex + offsetB;
+      const auto [offsetCell1, offsetCell2] = std::minmax(offsetCellA, offsetCellB);
 
-      auto cell2Local = globalToLocalIndex[offsetCell1].find(offsetCell2);
+      const auto cell2Local = globalToLocalIndex[offsetCell1].find(offsetCell2);
 
       // check if cell2 exists (in cell1's neighbor list)
       if (cell2Local != globalToLocalIndex[offsetCell1].end()) {
         // if aos, send every pair of interacting particles to functor
-        if (dataLayout == DataLayoutOption::aos) {
+        if constexpr (dataLayout == DataLayoutOption::aos) {
           // vector of pairs {particle, list}
-          auto &currentList = aosNeighborList[offsetCell1][cell2Local->second];
+          const auto &currentList = aosNeighborList[offsetCell1][cell2Local->second];
           for (auto &[particleBasePtr, particleList] : currentList) {
-            auto &particleBase = *particleBasePtr;
-            for (auto particlePartnerPtr : particleList) {
-              auto &particlePartner = *particlePartnerPtr;
-              pairwiseFunctor->AoSFunctor(particleBase, particlePartner, useNewton3);
+            for (auto *particlePartnerPtr : particleList) {
```

Comment:
Yes you are right, this is a `Particle *`. It will not make a difference if we put `*` because `auto` will resolve this correctly either way. It is [good practice](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-auto.html) though, to append `*` to highlight the fact that this is a pointer. Otherwise it might become complicated to deduce this.

Reply:
Ah, okay thanks for the link! Wasn't sure if the `*` was changing anything